### PR TITLE
Manual build trigger button (#45) resolved

### LINKED ISSUE #45: Manual build trigger button
Add a button to the repository page `repos/<user>/<repo>/hooks` to trigger a build manually at the current HEAD.  This will be useful for the first build when a user initially enables a hook and would like to test.

### PR DESCRIPTION
**Pull request by cervenkam**
Closes #45

The button is added directly to the template

### CODE REVIEW

File: internal/resources/templates/repolist.go [hunk 1]
```diff
@@ -84,6 +84,11 @@ const RepoPage = `
 					{{range $hookname, $hook := .Hooks}}
 						<tr>
 							<td class="name text bold four wide"><a href="">{{$hookname | ToUpper}}</a></td>
+							<td class="name four wide"><form action="/pubvalidate" method="post">
+								<input type="hidden" name="validator" value="{{$hookname}}" />
+								<input type="hidden" name="repopath" value="{{$.FullName}}" />
+								<input type="submit" value="Validate once" />
```
Comment (mpsonntag) [inline internal/resources/templates/repolist.go:90]: Please change text to "Run validation"

File: internal/resources/templates/repolist.go [hunk 2]
```diff
@@ -84,6 +84,11 @@ const RepoPage = `
 					{{range $hookname, $hook := .Hooks}}
 						<tr>
 							<td class="name text bold four wide"><a href="">{{$hookname | ToUpper}}</a></td>
+							<td class="name four wide"><form action="/pubvalidate" method="post">
```
Comment (mpsonntag) [inline internal/resources/templates/repolist.go:87]: Using the `/pubvalidate` route does not work with private repositories.

**Review by mpsonntag — CHANGES_REQUESTED** [review]
Using the `/pubvalidate` route does not work with private repositories.

Further the public validation does not link the `latest` validation and therefore a triggered validation run would not show up on the results page which is what the issue tries to solve.

Most probably a new route is required that checks the permissions internally, runs the required validation and links the results appropriately.

**Comment by cervenkam** [comment]
This PR is closed for now due to the fact that private repositories are in this state _hard to validate_.

**Pull request closed without merging**